six/modules/c++/six.sicd/include/six/sicd/ComplexData.h

```diff
@@ -334,6 +334,9 @@ class ComplexData: public Data
     }
 
     bool operator==(const ComplexData& rhs) const;
+    bool validate(logging::Logger& log) const;
```

Comment:
At least for now, just at this level... don't worry about doing it for the individual classes